src/cis/fam_cis_direct.cpp

```diff
@@ -173,11 +167,7 @@ void Fam_CIS_Direct::destroy_region(uint64_t regionId, uint64_t memoryServerId,
     // cached KVS is freed in metadata_delete_region and calling
     // metadata_delete_region after DestroyHeap will result in SIGSEGV.
 
-    ret = metadataManager->metadata_delete_region(regionId);
-    if (ret != META_NO_ERROR) {
-        message << "Can not remove region from metadata service";
-        throw CIS_Exception(REGION_NOT_REMOVED, message.str().c_str());
-    }
+    metadataManager->metadata_delete_region(regionId);
```

Comment:
Are you catching Metadata_Service_Exception (or its base class exception here or in callers of this routine (cis_server ? Or fam_allocator_client ? )

Reply:
I think you need to change cis_server to catch any exception. It catches CIS_Exception, So metadata_exception will not be caught.